### DISCUSSION TITLE
graphql support

### DISCUSSION THREAD

**Discussion by elhajjh** [oqtane/oqtane.framework#1977] [General]
Can GraphQL like Hotchocolate be used in Oqtane?

**Comment by sbwalker** [comment]
@elhajjh I assume what you are asking is if it is possible to use GraphQL in your custom modules you create in Oqtane (as the core framework is using standard REST APIs and does not plan to use GraphQL)? Basically your module would need to include the GraphQL DLLs and implement the IServerStartup interface to register the required startup services.

**Comment by elhajjh** [comment]
Hi
Thank you, you answered my exact question.

As an advice , do you recommend me using graphql/hotchocolate in my custom modules or rest api,
My custom modules will use alot of DB data.

Best Regards,






-------- Original message --------
From: Shaun Walker ***@***.***>
Date: 2/7/22 2:41 AM (GMT+02:00)
To: "oqtane/oqtane.framework" ***@***.***>
Cc: Elhajj ***@***.***>, Mention ***@***.***>
Subject: Re: [oqtane/oqtane.framework] graphql support (Discussion #1977)


@elhajjh<https://emea01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Felhajjh&data=04%7C01%7C%7C332dc6c8f35c47eb587f08d9e9d289d6%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637797912736132198%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=IJcyFY4AP4GMdGSUxXSS%2FnOlS79hnV7r7Vz4zbyLSd4%3D&reserved=0> I assume what you are asking is if it is possible to use GraphQL in your custom modules you create in Oqtane (as the core framework is using standard REST APIs and does not plan to use GraphQL)? Basically your module would need to include the GraphQL DLLs and implement the IServerStartup interface to register the required startup services.

—
Reply to this email directly, view it on GitHub<https://emea01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Foqtane%2Foqtane.framework%2Fdiscussions%2F1977%23discussioncomment-2122085&data=04%7C01%7C%7C332dc6c8f35c47eb587f08d9e9d289d6%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637797912736132198%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=SSEr%2B0U96Pqy4Qy7bh0AWYGNFLME5SjOnZ0mV3UjmeA%3D&reserved=0>, or unsubscribe<https://emea01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAAS7UAASMLJERZBJLBEVHYDUZ4IKPANCNFSM5NHTXPUQ&data=04%7C01%7C%7C332dc6c8f35c47eb587f08d9e9d289d6%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637797912736132198%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=YS669CdDOnRGtsZRWUezQ9zoUOdQ9TSqTxYcz48U%2BRI%3D&reserved=0>.
Triage notifications on the go with GitHub Mobile for iOS<https://emea01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapps.apple.com%2Fapp%2Fapple-store%2Fid1477376905%3Fct%3Dnotification-email%26mt%3D8%26pt%3D524675&data=04%7C01%7C%7C332dc6c8f35c47eb587f08d9e9d289d6%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637797912736132198%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=DlbBASGXb9P98sRkmNjwmKZ10b15PrbvrezUiJhW2Lg%3D&reserved=0> or Android<https://emea01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Dcom.github.android%26referrer%3Dutm_campaign%253Dnotification-email%2526utm_medium%253Demail%2526utm_source%253Dgithub&data=04%7C01%7C%7C332dc6c8f35c47eb587f08d9e9d289d6%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637797912736132198%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=1FPyAytiam5ko%2BrV8hqwaLBHOpImOaOdHklYmy0KOHk%3D&reserved=0>.

**Reply by sbwalker** [reply]
I do not know the business requirements for your module so I really cannot make any recommendations. Purely from the perspective of scaffolding your module I would recommend using the Module Creator (it is located on the Develop page in a new Oqtane installation) as it creates a fully functional module for you (with a REST API for the back end).
